cdap-elastic/src/test/java/io/cdap/cdap/metadata/elastic/ElasticsearchMetadataStorageTest.java

```diff
@@ -456,13 +456,54 @@ public void testOptionalTermsAreNotRequired() throws IOException {
     cleanMetadataStorage(records);
   }
 
+  @Test
+  public void testNumericQueries() throws IOException {
+    MetadataStorage mds = getMetadataStorage();
+    List<MetadataRecord> records = createRecordsWithMetadata();
+
+    // test each comparison operator
+    SearchRequest request = SearchRequest.of("p:=0").build();
+    SearchResponse response = mds.search(request);
+    Assert.assertEquals(1, response.getResults().size());
+    Assert.assertEquals(records.get(0), response.getResults().get(0));
+
+    SearchRequest request2 = SearchRequest.of("p:>0").build();
+    SearchResponse response2 = mds.search(request2);
+    Assert.assertEquals(2, response2.getResults().size());
+    Assert.assertEquals(records.get(1), response2.getResults().get(0));
+    Assert.assertEquals(records.get(2), response2.getResults().get(1));
+
+    SearchRequest request3 = SearchRequest.of("p:>=0").build();
+    SearchResponse response3 = mds.search(request3);
+    Assert.assertEquals(records, response3.getResults());
+
+    SearchRequest request4 = SearchRequest.of("p:<10").build();
+    SearchResponse response4 = mds.search(request4);
+    Assert.assertEquals(records, response4.getResults());
+
+    SearchRequest request5 = SearchRequest.of("p:<=1").build();
+    SearchResponse response5 = mds.search(request5);
+    Assert.assertEquals(2, response5.getResults().size());
+    Assert.assertEquals(records.get(0), response5.getResults().get(0));
+    Assert.assertEquals(records.get(1), response5.getResults().get(1));
+
+    // test multiple numeric bounds
+    SearchRequest request6 = SearchRequest.of("p:>=1 p:<3").build();
```

Comment:
Would this work if someone uses a syntax like `p:>=3.0`? If not, what changes would be required to support that?

Reply:
It would not. I hadn't thought to include that, since decimal property values can't be input. However, supporting this query would be as simple as changing Long.parseLong to Double.parseDouble wherever it's used in the query parsing process. To make things more generally applicable, it may be worth renaming SearchType.LONG to SearchType.NUMERIC.

Reply:
I address this in the latest commit by using Double where Long was previously used, and by renaming SearchType.LONG to SearchType.NUMERIC.